packages/admin/src/error/errorboundary/useErrorBoundary.ts

```diff
@@ -0,0 +1,13 @@
+import * as React from "react";
+
+const useErrorBoundary = () => {
```

Comment:
In my opinion, we shouldn't support the `useErrorBoundary` hook at all, as it promotes the use of exceptions for flow control...

Reply:
and not support throwing catchable error in event handlers at all?

Reply:
of course we can simple remove this hook and add it later if we need it. 

But i think it's not too bad and this hook just adds some more possibilities for error handling and does not really hurt someone or breaks anything.

On the other hand i am not really sure if someone will really use this kind of hook or error flow.

Reply:
My opinion might be too harsh due to personal preference, but when I consider the use case, the hook simply doesn't make sense to me: If a developer makes the effort to 1.) wrap the component in an `ErrorBoundary`, 2.) include the `useErrorBoundary` hook, 3. catch an error using try/catch and then "throw" said error, they might as well implement an actual error handling. 🤷🏼‍♂️

Reply:
However, there's _no actual harm_ in adding it (since avoiding exceptions for flow control is a personal preference), and as I am not forced to use it, it's okay for me. 🙂

Reply:
@johnnyomair: I think you are right, we don't need this hook. I can't think of an event handler I've written where it would have been useful.

Mainly because of uncaught exceptions in event handlers simply stop the event handler and display an error in the console. (In contrast to exception during render where would get a white screen)

And you would also loose context and get thrown back to the ErrorBoundy, which can cause loss of data. (Though, it would display an error, which currently doesn't happen)

Reply:
makes sense to me too and i will remove that part.